### DISCUSSION TITLE
Github Page url change

### DISCUSSION THREAD

**Discussion by GeoEngineer64** [community/community#23212] [Other Feature Feedback, Questions, & Ideas]
<p>I tried to change the repo name of my GitHub page repo in order to change the page’s url. Instead it just tacked on the new repo name to the old one as the page url. Even after deleting the repo, creating a new one with the desired name, and re-uploading the code, the page url is still doubled with the old one.</p>
<p>repo: <a href="https://github.com/GeoEngineer64/NikHawley.github.io" class="inline-onebox" rel="noopener nofollow ugc">GitHub - GeoEngineer64/NikHawley.github.io: Nik Hawley's Portfolio</a></p>
<p>Page: <a href="https://geoengineer64.github.io/NikHawley.github.io/" class="inline-onebox" rel="noopener nofollow ugc">Nik Hawley's Portfolio</a></p>

**Comment by tcbyrd** [comment, marked as answer]
<p>Hi there! The subdomain of <code>github.io</code> will always match your username. Every repository in your account is automatically hosted on <code>&lt;login&gt;.github.io</code> by default, and you cannot change that since it could conflict with another user. So if you want the <code>nikhawley.github.io</code> subdomain, you’d need an account with that username. The <code>&lt;username&gt;.github.io</code> repository has special treatment here, since it indicates to us that’s the site you want to be served at the root of your GitHub Pages domain. Any other repository in your account then gets hosted at <code>&lt;username&gt;.github.io/&lt;repo_name&gt;</code> by default, unless you add a custom domain name.</p>
<p>There’s more information on the types of GitHub Pages sites and now they relate to domains here: <a href="https://docs.github.com/en/free-pro-team@latest/github/working-with-github-pages/about-github-pages#types-of-github-pages-sites" class="inline-onebox" rel="noopener nofollow ugc">About GitHub Pages - GitHub Docs</a></p>
<p>Hope that helps!</p>

**Reply by muellert** [reply]
So how do I get a "repository.github.io" page? Because I think I've seen a bunch of those, and it'd be much nicer to not have your username clutter the url.

**Reply by SourLemonJuice** [reply]
They may have an organization for that: `<orgname>.github.io`

**Comment by GeoEngineer64** [comment]
<p>Gotcha. Thanks!</p>
<p>Based on my understanding at the time of the article I was reading about making a portfolio using GitHub pages, I thought it seemed like you could make the repo name <code>&lt;whatever&gt;.github.io</code> and as long as it had the <code>github.io</code>, that would be the url, but now I understand… thank you!</p>

**Comment by mastar123-prog** [comment]
Default DNS for GitHub Pages:
By default, GitHub Pages sites are hosted on a domain in the format: username.github.io/repository or username.github.io for user/organization sites. This is the default DNS that GitHub uses to point to your hosted site. 
Using a Custom Domain:
To use a custom URL (like your own domain name) instead of the default, you need to configure both your GitHub Pages repository settings and your DNS provider. 
Configuration Steps:
In your GitHub repository:
Navigate to your site's repository.
Go to Settings > Code and automation > Pages.
Under "Custom domain", type your custom domain and click Save.
This will create a CNAME file in your repository (if publishing from a branch).
With your DNS provider:
For apex domains (e.g., example.com): Create ALIAS, ANAME, or A records pointing to the GitHub Pages IP addresses.
For subdomains (e.g., www.example.com): Create a CNAME record that points to your default GitHub Pages domain (username.github.io or organization.github.io).
Important: Ensure the CNAME record points directly to your default GitHub Pages domain, without including the repository name. 
DNS Propagation:
DNS changes can take time to propagate across the internet, so your custom domain might not work immediately. It can take up to 24 hours. 
Verification:
You can verify your custom domain and ensure proper DNS configuration by using the dig command in the terminal. 
By following these steps, you can successfully use a custom URL to point to your GitHub Pages website, while still leveraging the default DNS infrastructure provided by GitHub.

**Reply by mastar123-prog** [reply]
> Default DNS for GitHub Pages: By default, GitHub Pages sites are hosted on a domain in the format: username.github.io/repository or username.github.io for user/organization sites. This is the default DNS that GitHub uses to point to your hosted site. Using a Custom Domain: To use a custom URL (like your own domain name) instead of the default, you need to configure both your GitHub Pages repository settings and your DNS provider. Configuration Steps: In your GitHub repository: Navigate to your site's repository. Go to Settings > Code and automation > Pages. Under "Custom domain", type your custom domain and click Save. This will create a CNAME file in your repository (if publishing from a branch). With your DNS provider: For apex domains (e.g., example.com): Create ALIAS, ANAME, or A records pointing to the GitHub Pages IP addresses. For subdomains (e.g., [www.example.com](http://www.example.com)): Create a CNAME record that points to your default GitHub Pages domain (username.github.io or organization.github.io). Important: Ensure the CNAME record points directly to your default GitHub Pages domain, without including the repository name. DNS Propagation: DNS changes can take time to propagate across the internet, so your custom domain might not work immediately. It can take up to 24 hours. Verification: You can verify your custom domain and ensure proper DNS configuration by using the dig command in the terminal. By following these steps, you can successfully use a custom URL to point to your GitHub Pages website, while still leveraging the default DNS infrastructure provided by GitHub.

You can also verify your custom domain and ensure proper DNS configuration by using the dig command in the terminal.
